### ISSUE TITLE
DirectML Model Builder and Architecture API

### ISSUE THREAD

**Issue by TheManInTheBox** [ahebert-lt/cx#272]
## Overview
Fluent API for defining neural network architectures using DirectML operators.

## Scope
Create a builder pattern API for composing neural network layers without external dependencies.

## Implementation
- DirectMLModelBuilder.cs with fluent interface
- Layer implementations (Dense, Embedding, Attention) using DirectML primitives
- Parameter initialization using DirectML random number generation
- Model compilation and graph optimization
- Export to custom CX consciousness model format

## API Design
`csharp
var model = new DirectMLModelBuilder()
    .AddDenseLayer(inputSize: 512, outputSize: 256, activation: ActivationType.ReLU)
    .AddDenseLayer(inputSize: 256, outputSize: 128, activation: ActivationType.ReLU)
    .AddOutputLayer(outputSize: 10, activation: ActivationType.Softmax)
    .Build();
`

## Technical Requirements
- Pure DirectML operator composition
- Memory layout optimization for DirectML tensors
- Integration with consciousness data types
- Custom serialization format

## Acceptance Criteria
- [ ] Build feedforward networks programmatically
- [ ] Support common layer types (Dense, Convolution, Activation)
- [ ] Initialize parameters with different strategies
- [ ] Validate model architecture
- [ ] Export/import models in CX format

## Dependencies
- DirectML 1.12+ APIs
- Existing CX type system
